app/src/main/java/waiter/App.java

```diff
@@ -36,23 +36,33 @@ private static Routes constructRoutes() {
         );
 
         routes.addRoute(
-                new Route("/simple_get_with_body", new Request.Method[]{Request.Method.GET}, okWithBodyHandler)
+                new Route("/simple_get_with_body", new Request.Method[]{Request.Method.GET}, okWithOnlyResponseBodyHandler)
         );
 
         routes.addRoute(
                 new Route("/head_request", new Request.Method[]{Request.Method.HEAD, Request.Method.OPTIONS}, okHandler)
         );
 
+        routes.addRoute(
+                new Route("/echo_body", new Request.Method[]{Request.Method.POST}, okWithRequestBodyHandler)
+        );
```

Comment:
Non-blocking: I think your design will need to change here for a future use case where the handler for two methods (for the same endpoint) need to be different.

Example: Consider an endpoint with POST and PUT methods. Suppose that the handler for the POST method should write data to some datastore and that the handler for the PUT method should update some data in a datastore. You _could_ handle the case I just described by bifurcating on the methods within your handler function, but I would urge you not to do that.